lib/index.js

```diff
@@ -424,61 +424,79 @@ ABI.stringify = function (types, values) {
   return ret
 }
 
+ABI.solidityHexValue = function (type, value, bitsize) {
+  // pass in bitsize = null if use default bitsize
+  var size, num
+  if (type.match('^[a-zA-Z0-9_]+(\\[[0-9]*\\])+$')) {
+    var sub_type = type.replace(/\[.*?\]/, '')
+    var array_values = value.map(function (v) {
```

Comment:
What happens if the type is a static-size array, like 'uint[2]'? I think it should assert that 'values.length' is of correct length here, and not ignore such mistake input.